packages/integration-sdk-cli/src/neo4j/neo4jGraphStore.ts

```diff
@@ -0,0 +1,168 @@
+      return result;
+    }
+    else {
+      const session = this.neo4jDriver.session({
+        database: this.databaseName,
+        defaultAccessMode: neo4j.session.WRITE,
+      });
+      const result = await session.run(cypherCommand);
+      await session.close();
+      return result;
+    }
+  }
+
+  private startsWithNumeric(str: string){
+    return /^\d+$/.test(str);
+  }
+
+  private sanitizePropertyName(propertyName: string): string {
+    let sanitizedName = '';
+    if(this.startsWithNumeric(propertyName)) {
+      sanitizedName += 'n';
+    }
+    sanitizedName += propertyName;
+    sanitizedName = sanitizedName.replace(/[\s!@#$%^&*()-=+\\|'";:/?.,><`~\t\n[\]{}]/g, "_");
+    return sanitizedName;
+  }
+
+  private sanitizeValue(value: string): string {
+    return value.replace(/'/gi, "\\'")
+  }
+
+  public buildPropertyString(propList, nodeName) {
+    let propString = '';
+    for (const key in propList) {
+      if (key === '_rawData') {
+        //stringify JSON in rawData so we can store it.
+        propString += `${nodeName}.${key} = '${JSON.stringify(propList[key])}', `;
+      } else {
+        // Sanitize out characters that aren't allowed in property names
+        const propertyName = this.sanitizePropertyName(key);
+
+        //If we're dealing with a number or boolean, leave alone, otherwise
+        //wrap in single quotes to convert to a string and escape all
+        //other single quotes so they don't terminate strings prematurely.
+        if(typeof propList[key] == 'number' || typeof propList[key] == 'boolean') {
+          propString += `${nodeName}.${propertyName} = ${propList[key]}, `;
+        }
+        else {
+          propString += `${nodeName}.${propertyName} = '${this.sanitizeValue(propList[key].toString())}', `;
+        }
+      }
+    }
+    propString = propString.slice(0, -2);
+
+    return propString;
+  }
+
+  async addEntities(newEntities: Entity[]) {
+    for (const entity of newEntities) {
+      //Add index if not already in types.  This will optimize future
+      //MATCH/MERGE calls.
+      if (!this.typeList.has(entity._type)) {
+        await this.runCypherCommand(
+          `CREATE INDEX index_${entity._type} IF NOT EXISTS FOR (n:${entity._type}) ON (n._key, n._integrationInstanceID);`,
+        );
+        this.typeList.add(entity._type);
+      }
+      const propString = this.buildPropertyString(entity, 'n');
+      const finalKeyValue = this.sanitizeValue(entity._key.toString());
+      const buildCommand = `
+        MERGE (n {_key: '${finalKeyValue}', _integrationInstanceID: '${this.integrationInstanceID}'}) 
+        SET ${propString}, n:${entity._type};`;
+      await this.runCypherCommand(buildCommand);
+    }
+  }
+
+  async addRelationships(newRelationships: Relationship[]) {
+    for (const relationship of newRelationships) {
+      const propString = this.buildPropertyString(relationship, 'relationship');
```

Comment:
I see that you chose `relationship` here and `n` above

Reply:
Yep.  Wanted to denote between the two, since one is a node alias we're using for property value sets and this is a relationship alias.  I'll pull it out into a const to clean it up a bit though.

Reply:
Updated in latest commit.